Sources/SWBUniversalPlatform/TestEntryPointGenerationTool.swift

```diff
@@ -17,7 +17,59 @@ import SWBCore
 final class TestEntryPointGenerationToolSpec: GenericCommandLineToolSpec, SpecIdentifierType, @unchecked Sendable {
     static let identifier = "org.swift.test-entry-point-generator"
 
+    override func commandLineFromTemplate(_ cbc: CommandBuildContext, _ delegate: any TaskGenerationDelegate, optionContext: (any DiscoveredCommandLineToolSpecInfo)?, specialArgs: [String] = [], lookup: ((MacroDeclaration) -> MacroExpression?)? = nil) -> [CommandLineArgument] {
+        var args = super.commandLineFromTemplate(cbc, delegate, optionContext: optionContext, specialArgs: specialArgs, lookup: lookup)
+        for (toolchainPath, toolchainLibrarySearchPath) in cbc.producer.toolchains.map({ ($0.path, $0.librarySearchPaths) }) {
+            if let path = toolchainLibrarySearchPath.findLibrary(operatingSystem: cbc.producer.hostOperatingSystem, basename: "IndexStore") {
+                args.append(contentsOf: ["--index-store-library-path", .path(path)])
+            }
+            for input in cbc.inputs {
+                if input.fileType.conformsTo(identifier: "text") {
+                    args.append(contentsOf: ["--linker-filelist", .path(input.absolutePath)])
+                } else if input.fileType.conformsTo(identifier: "compiled.mach-o") {
+                    // Do nothing
+                } else {
+                    delegate.error("Unexpected input of type '\(input.fileType)' to test entry point generation")
+                }
+            }
+        }
+        return args
+    }
+
     override func createTaskAction(_ cbc: CommandBuildContext, _ delegate: any TaskGenerationDelegate) -> (any PlannedTaskAction)? {
         TestEntryPointGenerationTaskAction()
     }
+
+    public func constructTasks(_ cbc: CommandBuildContext, _ delegate: any TaskGenerationDelegate, indexStorePaths: [Path], indexUnitBasePaths: [Path]) async {
```

Comment:
Can we inject the necessary additional arguments without overriding constructTasks? Ever bare call to createTasks adds risk that we forget to update things as its signature or default behavior changes.

Reply:
I tried to do this at first, but I couldn't find an elegant way to differentiate the input types, especially since the handling of the index store is a bit unusual since the build system doesn't track the content inside it